packages/frontend/editor-ui/src/components/AskAssistant/Agent/AskAssistantBuild.vue

```diff
@@ -94,6 +99,33 @@ watch(
 	{ deep: true },
 );
 
+// If this is the initial generation, streaming has ended, and there were workflow updates,
+// we want to save the workflow
+watch(
+	() => builderStore.streaming,
+	async () => {
+		if (
+			builderStore.initialGeneration &&
+			!builderStore.streaming &&
+			workflowsStore.workflow.nodes.length > 0
+		) {
+			// Check if the generation completed successfully (no error or cancellation)
+			const lastMessage = builderStore.chatMessages[builderStore.chatMessages.length - 1];
+			const successful =
+				lastMessage &&
+				lastMessage.type !== 'error' &&
+				!(lastMessage.type === 'text' && lastMessage.content === '[Task aborted]');
```

Comment:
Nit: Would be nice to have this as a const since it's now used here and builder.store

Reply:
Would it make sense to make it a message of type 'error'? Or add separate message type 'aborted'. I don't like logic based on captions either :)